File: Memento/python/Editor.py
from EditorState import EditorState


class Editor:
    __content = str()

    def createState(self) -> EditorState:
        return EditorState(self.__content)

    def restore(self, state: EditorState) -> None:
        self.__content = state.getContent()

    def getContent(self):
        return self.__content

    def setContent(self, content: str):
        self.__content = content

File: Memento/dart/Editor.dart
import 'EditorState.dart';

class Editor {
  late String _content;

  EditorState createState() {
    return EditorState(content: this._content);
  }

  void restore(EditorState state) {
    this._content = state.content;
  }

  String get content => _content;

  set content(String content) {
    _content = content;
  }
}
